skimage/measure/_colocalization.py

```diff
@@ -0,0 +1,371 @@
+import numpy as np
+from scipy.stats import pearsonr
+from functools import wraps
+from .._shared.utils import check_shape_equality
+
+__all__ = ['pearson_r',
+           'manders_colocalization_coeff',
+           'manders_overlap_coeff',
+           'overlap',
+           'integrated_density',
+           'average_integrated_density'
+           ]
+
+
+def check_shape_equality_all(*args):
+    # check that all np.arrays passed to it have the same shape
+    im1 = args[0]
+    for im in args[1:]:
+        check_shape_equality(im1, im)
+    return
+
+
+def check_numpy_arr(arr, name, bool_expected=False):
```

Comment:
I would just remove this check utility altogether. Perhaps just call something like

```Python
image = np.asarray(image)
```
for images and 
```
mask = np.asarray(mask, dtype=bool)
```
for masks

`asarray` will not make a copy if the input is already an array of the expected dtype

Reply:
I made this change for image arguments. For masks, I wanted to raise a warning if a non-binary mask was inputted (e.g. an intensity image layer) because if this is done accidently, the metric would be calculated without the user realising. I added the function `is_binary_ndarray()` in `skimage._shared.utils` to do that.